src/Widgets/VPN/VPNMenuItem.vala

```diff
@@ -16,106 +16,135 @@
  */
 
 public class Network.VPNMenuItem : Gtk.ListBoxRow {
-    public signal void user_action ();
-    public NM.RemoteConnection? connection { get; set; }
+    public signal void connect_clicked ();
+    public signal void disconnect_clicked ();
+    public NM.RemoteConnection? connection { get; construct; }
 
     public Network.State state { get; set; default = Network.State.DISCONNECTED; }
 
-    Gtk.RadioButton radio_button;
-    Gtk.Spinner spinner;
-    Gtk.Image error_img;
-    Gtk.Button remove_button;
+    private static Gtk.SizeGroup size_group;
 
-    public VPNMenuItem (NM.RemoteConnection _connection, VPNMenuItem? previous = null) {
-        Object (connection: _connection);
+    private Gtk.Button connect_button;
+    private Gtk.Image vpn_state;
+    private Gtk.Label state_label;
+    private Gtk.Label vpn_label;
+    private Gtk.LinkButton vpn_info_button;
 
-        var main_box = new Gtk.Box (Gtk.Orientation.HORIZONTAL, 6);
-        main_box.margin_start = main_box.margin_end = 6;
-        radio_button = new Gtk.RadioButton (null);
-        if (previous != null) {
-            radio_button.set_group (previous.radio_button.get_group ());
-        }
+    private string state_label_text;
 
-        error_img = new Gtk.Image.from_icon_name ("process-error-symbolic", Gtk.IconSize.MENU);
-        error_img.set_tooltip_text (_("This Virtual Private Network could not be connected to."));
+    public VPNMenuItem (NM.RemoteConnection _connection) {
+        Object (
+            connection: _connection
+        );
+    }
 
-        spinner = new Gtk.Spinner ();
-        spinner.visible = false;
-        spinner.no_show_all = !spinner.visible;
+    static construct {
+        size_group = new Gtk.SizeGroup (Gtk.SizeGroupMode.HORIZONTAL);
+    }
 
-        remove_button = new Gtk.Button.from_icon_name ("user-trash-symbolic", Gtk.IconSize.MENU);
-        remove_button.get_style_context ().add_class ("flat");
-        remove_button.clicked.connect (() => {
-            try {
-                connection.delete (null);
-            } catch (Error e) {
-                warning (e.message);
-            }
+    construct {
+        var image = new Gtk.Image.from_icon_name ("network-vpn", Gtk.IconSize.DND);
+
+        vpn_state = new Gtk.Image.from_icon_name ("user-offline", Gtk.IconSize.MENU);
+        vpn_state.halign = Gtk.Align.END;
+        vpn_state.valign = Gtk.Align.END;
+
+        state_label = new Gtk.Label (null);
+        state_label.xalign = 0;
+        state_label.use_markup = true;
+
+        var overlay = new Gtk.Overlay ();
+        overlay.add (image);
+        overlay.add_overlay (vpn_state);
+
+        vpn_label = new Gtk.Label (connection.get_id ());
+        vpn_label.ellipsize = Pango.EllipsizeMode.END;
+        vpn_label.hexpand = true;
+        vpn_label.xalign = 0;
+
+        vpn_info_button = new Gtk.LinkButton ("");
```

Comment:
Is there a reason for this to be a link button?

Reply:
Just wanted to be consistent with Bluetooth plug here.

Reply:
Made it a `Gtk.Button`